### ISSUE TITLE
Option to Display a Man page on --help

### ISSUE THREAD

**Issue by nhumrich** [datafolklabs/cement#291]
Its common in CLI's for only the base command to print the standard help menu. All subcommands typically show a man page on the when the --help option is present. For example

    git --help
    git add --help

Using git add gives you a detailed man page. It would be nice to have the ability to show a man page for specific subcommands. 

I am not so much conserned about the creation of a man page, just the showing of one. I would initially be willing to create my manpages manually.

**Comment by derks** [comment]
Thanks for the feedback. I like the idea... will probably implement as a more arbitrary means of overriding what happens when `--help` is passed... rather than being specific to calling Man... but I think we can make something pretty easy/seamless to use.

**Comment by derks** [comment]
I've worked up an example of how to do this in Cement.  Note that this is using new features available in `master` and will be out in Cement 2.8:

```python

import os
import argparse
from cement.core.foundation import CementApp
from cement.ext.ext_argparse import ArgparseArgumentHandler
from cement.ext.ext_argparse import ArgparseController, expose

class BaseController(ArgparseController):
    class Meta:
        label = 'base'
        arguments = [
            (['--help'], dict(help='custom help', action='store_true')),
        ]

    def _post_argument_parsing(self):
        if self.app.pargs.help:
            os.system('man myapp')
            self.app.close(0)

    @expose(hide=True)
    def default(self):
        print('Inside BaseController.default')

class MyApp(CementApp):
    class Meta:
        label = 'myapp'
        argument_handler = ArgparseArgumentHandler(add_help=False)
        exit_on_close = True
        handlers = [BaseController]


with MyApp() as app:
    app.run()
```

Few things to note:
 
 * Override the `argument_handler` in order to pass `add_help=False` to the `ArgparseArgumentHandler`
 * Add a new `--help` option to the base controller
 * Handle the `--help` option in `_post_argument_parsing` on the base controller, and shell out to `man`


Looks like:

```
$ python myapp.py
Inside BaseController.default


$ python myapp.py --help
No manual entry for myapp
```

Obviously, I don't have a man page for `myapp` but you see where it's going.  Hope that helps.
